nginx-controller/controller/controller.go

```diff
@@ -898,9 +915,22 @@ func (lbc *LoadBalancerController) createIngress(ing *extensions.Ingress) (*ngin
 		}
 	}
 
+	if mergeableType, exists := ingEx.Ingress.Annotations["nginx.org/mergible-ingress-type"]; exists {
```

Comment:
Better not to change `createIngress` method at all

Reply:
I think it is better not to change createIngress at all and validate masters and minions before calling createIngress